process/smartContract/builtInFunctions/saveUserName.go

```diff
@@ -45,34 +43,35 @@ func NewSaveUserNameFunc(
 func (s *saveUserName) ProcessBuiltinFunction(
 	_, acntDst state.UserAccountHandler,
 	vmInput *vmcommon.ContractCallInput,
-) (*big.Int, uint64, error) {
+) (*vmcommon.VMOutput, error) {
 	if vmInput == nil {
-		return big.NewInt(0), 0, process.ErrNilVmInput
+		return nil, process.ErrNilVmInput
+	}
+	if vmInput.GasProvided < s.gasCost {
+		return nil, process.ErrNotEnoughGas
 	}
 	if check.IfNil(acntDst) {
-		return big.NewInt(0), vmInput.GasProvided, process.ErrNilSCDestAccount
+		// cross-shard call, in sender shard only the gas is taken out
+		return &vmcommon.VMOutput{ReturnCode: vmcommon.Ok}, nil
```

Comment:
It is not possible to call this cross-shard. as the calledAddress has to be equal with the acntDST. So it is only in shard. But instead of returning a stupid error for a bad call, I would just consume all the gas here.

Reply:
returned vmcommon user error